src/main/java/com/iemr/hwc/fhir/provider/condition/ConditionExtProvider.java

```diff
@@ -59,4 +73,65 @@ public MethodOutcome createCondition(HttpServletRequest theRequest, @ResourcePar
         method.setResource(conditionService.addNewChiefComplaint(theRequest,conditionExt));
         return method;
     }
+
+    @Search()
+    public List<ConditionExt> findChiefComplaintByDistrictAndLastModifDate(HttpServletRequest theRequest, @RequiredParam(name = "providerServiceMapId") StringParam providerServiceMapId, @RequiredParam(name = "vanID") StringParam vanID, @RequiredParam(name = "lastModif") DateParam lastModifyDate) {
+
+        List<ConditionExt> listRes = new ArrayList<>();
+        try {
+            String authorization = theRequest.getHeader("Authorization");
+            List<BenChiefComplaint> listChiefComplaint = conditionServiceImpl.getChiefComplaintByLocationAndLastModifDate(Integer.parseInt(providerServiceMapId.getValue()), Integer.parseInt(vanID.getValue()),
+                    new Timestamp(lastModifyDate.getValue().getTime()));
+
+            for (BenChiefComplaint benef:listChiefComplaint) {
+                BeneficiaryFlowStatus beneficiaryFlowStatus = beneficiaryFlowStatusRepo.getBenFlowByVisitIDAndVisitCode(benef.getBenVisitID(), benef.getVisitCode());
+                ConditionExt condition = new ConditionExt();
+                condition.setId(benef.getBenChiefComplaintID()+"");
+                condition.setProviderServiceMapId(new StringType(benef.getProviderServiceMapID()+""));
+                condition.setVanID(new StringType(benef.getVanID()+""));
+                condition.setParkingPlaceID(new StringType(benef.getParkingPlaceID()+""));
+                condition.setCreatedBy(new StringType(benef.getCreatedBy()));
+                condition.setBeneficiaryRegID(new StringType(benef.getBeneficiaryRegID()+""));
+                if (beneficiaryFlowStatus != null){
+                    condition.setBeneficiaryID(new StringType(beneficiaryFlowStatus.getBeneficiaryID()+""));
+                    condition.setBenFlowID(new StringType(beneficiaryFlowStatus.getBenFlowID()+""));
+                }
+                else {
+                    throw new ResourceNotFoundException("No record found for given benVisitID and BenVisitCode");
```

Comment:
Instead of throwing exception, log it as no benVisitID and BenvisitCode. Add continue statement to ignore the record. Can we write in the query to get only records with benVistiID and BenvisitCode?

Reply:
I even think that we don't need to generate an exception because the benVisitID and the visitCode will exist all the time. So I removed the generation of an exception @roopesh-beehyv